testing/test-utils/src/lib/utils/os-agnostic-paths.ts

```diff
@@ -43,6 +43,7 @@ const OS_AGNOSTIC_CWD = `<CWD>`;
  * Output: ../my-folder/my-file.ts
  *
  */
+export function osAgnosticPath(filePath: string): string;
```

Comment:
Why add this function declaration?

Reply:
To get better return type narrowing. (thx btw for looking at it, i forgot the one for undefined :) )

**Before:** 
<img width="583" alt="Screenshot 2024-08-08 at 17 17 57" src="https://github.com/user-attachments/assets/f3e96bdd-fc05-41dc-82df-4b5fb9ea41c4">

**After:** 

<img width="583" alt="Screenshot 2024-08-08 at 17 18 06" src="https://github.com/user-attachments/assets/a75e8b41-5fbc-404b-bda1-32c0f6ac8b6a">

Reply:
I see. Yes, with both overloads it makes sense :+1: